pkgs/applications/networking/instant-messengers/ricochet-refresh/default.nix

```diff
@@ -0,0 +1,84 @@
+{ makeDesktopItem
+, lib
+, stdenv
+, fetchurl
+, fontconfig
+, freetype
+, libxcb
+, libxkbcommon
+, xorg
+}:
+
+assert stdenv.hostPlatform.system == "x86_64-linux";
+
+let
+  version = "3.0.10";
+  rpath = lib.makeLibraryPath [
+    fontconfig
+    freetype
+    libxcb
+    libxkbcommon
+    stdenv.cc.cc.lib
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+    xorg.libX11
+    xorg.libXext
+  ];
+  desktopItem = makeDesktopItem {
+    name = "ricochet-refresh";
+    exec = "ricochet-refresh";
+    icon = "ricochet-refresh";
+    desktopName = "Ricochet Refresh";
+    genericName = "Ricochet Refresh";
+    categories = "Network;InstantMessaging;";
+  };
+
+in stdenv.mkDerivation {
+  name = "ricochet-refresh-${version}";
+
+  # nix by default strips binaries (to save space?) but there is
+  # an outstanding issue where this breaks if rpath is set to a longer value
+  # using patchelf prior
+  # see: https://github.com/NixOS/patchelf/issues/10
+  dontStrip = true;
+
+  src = fetchurl {
+    url = "https://github.com/blueprint-freespeech/ricochet-refresh/releases/download/v${version}-release/ricochet-refresh-${version}-linux-x86_64.tar.gz";
```

Comment:
Can we use the github tarball instead?

Reply:
No, this is a binary release

Reply:
Is there a reason why we are not building from source here?

Reply:
The build system/scripts aren't exactly stable, binary release makes it easier for us to maintain.

Reply:
According to their docs https://github.com/blueprint-freespeech/ricochet-refresh/blob/main/BUILDING.md#qt-sdk it looks like plain standard qmake.

Reply:
> looks like plain standard qmake

yepp, see my [ninja edit](https://github.com/NixOS/nixpkgs/pull/128440#issuecomment-903491817)

Reply:
@pospeselr fixed : )